src/lib/runtime_config/runtime_config.ml

```diff
@@ -450,12 +450,20 @@ module Json_layout = struct
     let of_yojson json = of_yojson_generic ~fields of_yojson json
   end
 
+  module Data_hashes = struct
+    type ledger_with_hash = { merkle_root : string; s3_hash : string }
+    [@@deriving yojson, dhall_type]
+
+    type t = ledger_with_hash list [@@deriving yojson, dhall_type]
+  end
+
   type t =
     { daemon : Daemon.t option [@default None]
     ; genesis : Genesis.t option [@default None]
     ; proof : Proof_keys.t option [@default None]
     ; ledger : Ledger.t option [@default None]
     ; epoch_data : Epoch_data.t option [@default None]
+    ; s3_data_hashes : Data_hashes.t option [@default None]
```

Comment:
Can we add `ledger_with_hash` as part of `Ledger.t` as it corresponds to each ledger and is applicable to all variants - `Accounts | Named | Hash`?

Nit: Merkle root is redundant in the case of `Hash`. So additionally maybe something like the following to avoid that (we can ignore it if it increases complexity/confusion on the processing/checking side)
```
  type ledger_with_hash = { merkle_root : string; s3_hash : string }
  type base =
    | Named of string  * ledger_with_hash (** One of the named ledgers in [Genesis_ledger] *)
    | Accounts of Accounts.t  * ledger_with_hash (** A ledger generated from the given accounts *)
    | Hash of string  * ledger_with_hash (** The ledger with the given root hash *)
  [@@deriving bin_io_unversioned]
```
```

Reply:
Done that, good call, makes the code SO much nicer. To deal with the redundancy I deleted the `of string` constructor data for `Hash` and simply look it up in the parent field when needed.